File: AirlineReservation/src/app/flight-details.ts
import { Time } from "@angular/common";

export class FlightDetails {
	
	
	public flightid:number;
	public arrivalTime:Time;
	public businessClassPrice:number;
	public cabin:string;
	public capacity:number;
	public departureTime:Time;
	public destination:string;
	public duration:Time;
	public economyClassPrice:number;
	public source:string;

	public Flightdetails() {
	}

	public getFlightid():number {
		return this.flightid;
	}

	public setFlightid(flightid:number) {
		this.flightid = flightid;
	}

	public getArrivalTime():Time {
		return this.arrivalTime;
	}

	public setArrivalTime(arrivalTime:Time) {
		this.arrivalTime = arrivalTime;
	}

	public getBusinessClassPrice() {
		return this.businessClassPrice;
	}

	public setBusinessClassPrice(businessClassPrice:number) {
		this.businessClassPrice = businessClassPrice;
	}

	public getCabin():string {
		return this.cabin;
	}

	public setCabin(cabin:string) {
		this.cabin = cabin;
	}

	public getCapacity():number {
		return this.capacity;
	}

	public setCapacity(capacity:number) {
		this.capacity = capacity;
	}

	public getDepartureTime():Time {
		return this.departureTime;
	}

	public setDepartureTime(departureTime) {
		this.departureTime = departureTime;
	}

	public getDestination():string {
		return this.destination;
	}

	public setDestination(destination:string) {
		this.destination = destination;
	}

	public getDuration():Time {
		return this.duration;
	}

	public setDuration(duration:Time) {
		this.duration = duration;
	}

	public getEconomyClassPrice():number {
		return this.economyClassPrice;
	}

	public setEconomyClassPrice(i:number) {
		this.economyClassPrice = i;
	}

	public getSource():string {
		return this.source;
	}

	public setSource(source:string) {
		this.source = source;
	}
}

File: AirlineProject/src/main/java/lti/project/backend/Pojos/FlightDetails.java
package lti.project.backend.Pojos;

import java.io.Serializable;
import javax.persistence.*;

import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonFormat.Shape;
import com.fasterxml.jackson.annotation.JsonIgnore;

import java.sql.Timestamp;
import java.util.List;


@Entity
@Table(name="FLIGHTDETAILS")
public class Flightdetails implements Serializable {
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue
	private int flightid;

	@Column(name="ARRIVAL_TIME")
	@JsonFormat(shape = Shape.STRING,pattern="yyyy-MM-dd'T'HH:mm:ss",timezone="Asia/Kolkata")
	private Timestamp arrivalTime;

	@Column(name="BUSINESS_CLASS_PRICE",precision=1)
	private int businessClassPrice;

	@Column(length=1)
	private String cabin;

	@Column(length=2)
	private int capacity;

	@Column(name="DEPARTURE_TIME",length=20)
	@JsonFormat(shape = Shape.STRING,pattern="yyyy-MM-dd'T'HH:mm:ss",timezone="Asia/Kolkata")
	private Timestamp departureTime;

	@Column(length=10)
	private String destination;

	@JsonFormat(shape = Shape.STRING,pattern="yyyy-MM-dd'T'HH:mm:ss",timezone="Asia/Kolkata")
	private Timestamp duration;

	@Column(name="ECONOMY_CLASS_PRICE",precision=1)
	private int economyClassPrice;
	
	@Column(length=10)
	private String source;

	//bi-directional many-to-one association to Bookingdetail
	@OneToMany(mappedBy="flightdetail",fetch =FetchType.LAZY,cascade = CascadeType.MERGE)
	private List<Bookingdetails> bookingdetails;

	public Flightdetails() {
	}

	public int getFlightid() {
		return this.flightid;
	}

	public void setFlightid(int flightid) {
		this.flightid = flightid;
	}

	public Timestamp getArrivalTime() {
		return this.arrivalTime;
	}

	public void setArrivalTime(Timestamp arrivalTime) {
		this.arrivalTime = arrivalTime;
	}

	public int getBusinessClassPrice() {
		return this.businessClassPrice;
	}

	public void setBusinessClassPrice(int businessClassPrice) {
		this.businessClassPrice = businessClassPrice;
	}

	public String getCabin() {
		return this.cabin;
	}

	public void setCabin(String cabin) {
		this.cabin = cabin;
	}

	public int getCapacity() {
		return this.capacity;
	}

	public void setCapacity(int capacity) {
		this.capacity = capacity;
	}

	public Timestamp getDepartureTime() {
		return this.departureTime;
	}

	public void setDepartureTime(Timestamp departureTime) {
		this.departureTime = departureTime;
	}

	public String getDestination() {
		return this.destination;
	}

	public void setDestination(String destination) {
		this.destination = destination;
	}

	public Timestamp getDuration() {
		return this.duration;
	}

	public void setDuration(Timestamp duration) {
		this.duration = duration;
	}

	public int getEconomyClassPrice() {
		return this.economyClassPrice;
	}

	public void setEconomyClassPrice(int i) {
		this.economyClassPrice = i;
	}

	@JsonIgnore
	public List<Bookingdetails> getBookingdetails() {
		return bookingdetails;
	}

	public void setBookingdetails(List<Bookingdetails> bookingdetails) {
		this.bookingdetails = bookingdetails;
	}

	public String getSource() {
		return this.source;
	}

	public void setSource(String source) {
		this.source = source;
	}

}